### PR TITLE
Config debug comments

### PR DESCRIPTION
**Pull request by DG12**
Including not enabling PHOTO,NTC and DPS310

### CODE REVIEW

File: src/application_config/application_mode_debug.h [hunk 1]
```diff
@@ -1,20 +1,43 @@
...
+/* Save history log every second rather than the default 5 minutes  */
+/* This will result in writing FLASH log block every 7.36 minutes ie: 07:21 */
 #define APP_LOG_INTERVAL_S (1U) //!< Gets limited to heartbeat rate.
 
-#endif
\ No newline at end of file
+/* Not enough flash to load debug with all drivers so don't enable rarely needed ones */
+#define APP_SENSOR_PHOTO_ENABLED  (false)    
+#define APP_SENSOR_NTC_ENABLED    (false)
+#define APP_SENSOR_DPS310_ENABLE  (false)
```
Comment (ojousima) [inline src/application_config/application_mode_debug.h:41]: DPS310 is used in RuuviTag B8 boards

File: test/test_app_log.c [hunk 2]
```diff
@@ -818,7 +823,7 @@ void test_app_log_process_nomem_block (void)
     err_code = app_log_process (&sample);
     TEST_ASSERT (RD_SUCCESS == err_code);
 }
-
+#if     0 // unused functions should not be tested          \/  \/  \/
```
Comment (ojousima) [inline test/test_app_log.c:826]: Remove

File: test/test_app_log.c [hunk 3]
```diff
@@ -872,7 +877,7 @@ void test_app_log_config_set_notinit (void)
     err_code |= app_log_config_set (&defaults);
     TEST_ASSERT (RD_ERROR_INVALID_STATE == err_code);
 }
-
+#endif  //  unused functions should not be tested    /\    /\   /\
```
Comment (ojousima) [inline test/test_app_log.c:880]: Remove

File: src/application_config/application_mode_debug.h [hunk 4]
```diff
@@ -1,20 +1,43 @@
...
+#define SEGGER_RTT_CONFIG_DEFAULT_MODE (1U) 
+
 #define APP_WDT_INTERVAL_MS (10U*60U*1000U)
 
 /** @brief Communicate sensor data at this interval. 221 matches Apple guideline. */
 #define APP_BLE_INTERVAL_MS (221U)
 
 /** @brief repeat same data N times in advertisement, reduce sensor reads. */
-#define APP_NUM_REPEATS 4
+#define APP_NUM_REPEATS (4U)
```
Comment (ojousima) [inline src/application_config/application_mode_debug.h:29]: Thanks, this is better

File: src/application_config/application_mode_debug.h [hunk 5]
```diff
@@ -1,20 +1,43 @@
...
+ * @copyright Ruuvi Innovations Ltd, license BSD-3-Clause.
+ *
+ *  EDIT THIS file to customize
+ */ 
 #ifndef APPLICATION_MODE_DEBUG_H
 #define APPLICATION_MODE_DEBUG_H
 
 #define APP_FW_VARIANT "+debug"
-#define RI_LOG_ENABLED 1
+#define RI_LOG_ENABLED (true)
```
Comment (ojousima) [inline src/application_config/application_mode_debug.h:16]: (1U)

File: src/application_config/application_mode_default.h [hunk 6]
```diff
@@ -17,10 +21,6 @@
 #define APP_FW_NAME "Ruuvi FW "
 #endif
 
-#ifndef APP_FW_VERSION
-#define APP_FW_VERSION "v0.0.1"
```
Comment (ojousima) [inline src/application_config/application_mode_default.h:21]: Ok to move this to "higher level" config file, thanks

File: src/application_config/application_mode_default.h [hunk 7]
```diff
@@ -1,14 +1,18 @@
-#ifndef APPLICATION_MODE_DEFAULT_H
-#define APPLICATION_MODE_DEFAULT_H
```
Comment (ojousima) [inline src/application_config/application_mode_default.h:2]: Keep include guard

File: src/application_config/application_mode_default.h [hunk 8]
```diff
@@ -62,4 +62,5 @@
 #ifndef APP_LOCKED_AT_BOOT
 #   define APP_LOCKED_AT_BOOT (true)
 #endif
-#endif
```
Comment (ojousima) [inline src/application_config/application_mode_default.h:65]: Keep include guard

**Review by ojousima — CHANGES_REQUESTED** [review]
Please see comments, also app_log.c merge conflict needs to be resolved.

File: test/test_app_log.c (hunk 2 above)
Comment (DG12) [reply test/test_app_log.c:826]: I thought we agreed to cancel this previously ?
Is there something I (we) need to do to convince git to stop attempting to update test_app_log.c ??

File: test/test_app_log.c (hunk 3 above)
Comment (DG12) [reply test/test_app_log.c:880]: I thought we agreed to cancel this previously ?

File: src/application_config/application_mode_debug.h (hunk 1 above)
Comment (DG12) [reply src/application_config/application_mode_debug.h]: OOPS.
Un-"falsed" DPS310

File: src/application_config/application_mode_default.h (hunk 7 above)
Comment (DG12) [reply src/application_config/application_mode_default.h:2]: I didn't **remove**  
APPLICATION_MODE_DEFAULT_H
only  **moved** it up

File: src/application_config/application_mode_debug.h (hunk 5 above)
Comment (DG12) [reply src/application_config/application_mode_debug.h:16]: booleans like xxxxxxx_ENABLED should be defined as (true).  Right?

**Comment by DG12** [comment]
Is there something I need to do?

**Pull request closed without merging**